pandas/core/series.py

```diff
@@ -4939,6 +4939,16 @@ def _replace_single(self, to_replace, method: str, inplace: bool, limit):
     # error: Cannot determine type of 'shift'
     @doc(NDFrame.shift, klass=_shared_doc_kwargs["klass"])  # type: ignore[has-type]
     def shift(self, periods=1, freq=None, axis=0, fill_value=None) -> Series:
```

Comment:
The return type should be `Series | DataFrame`. Making this change might cause other typing issues; e.g. if the code expects getting back a Series when calling `.shift(1)`, mypy will no longer be able to deduce this function is returning a Series even though it is.

If this occurs, you can type-hint `periods` as `int | Collection[int]` (collection is a sized iterable) and then use overloads. This way mypy will be able to deduce when it gets back a Series vs a DataFrame based on the input.

Reply:
Thank you for the help with this! I will change the expected to Series | DataFrame for shift and also type-hint periods.

If you have the time to answer this, does the "CI / checks" check use the type-hints to verify what type is being returned vs what should be returned?

Reply:
It uses mypy. You should be able to run this yourself by going to the package directory and running `mypy pandas` from the command line.